src/api/crypto/eliptic_curve/Common.cpp

```diff
@@ -15,4 +15,13 @@ std::string_view get_pubkey_prefix(
                                           : not_correct_result;
 }
 
+std::string add_leading_zeros(
+    std::string const& value,
+    unsigned int desiredSize)
+{
+    const unsigned int leading_zeros_number = desiredSize - value.size();
+    std::string leading_zeros(leading_zeros_number, '0');
+    return leading_zeros + value;
```

Comment:
use join for concatenating strings, this part of algorithm is header only in librpc common

Reply:
we don't have librpc in opentxs

Reply:
Oh, I need to always first check to which repo PR is related to :D anyway as far as I remember you added to opentxs the same function with different name right ?

Reply:
We copied this join method from the librpc into opentxs, so we can concatenate string. That's the "internal/util/StringJoiner.hpp" header. But in this example, it is not necessary to include this header. We can create an oneliner with `+` concatenation:
```
return std::string(desired_size - value.size(), '0') + value;
```

Reply:
but we decided with Bartek that oneliner is not needed because multiple lines give extra information for code reader and have no negative effect on performance.